core/cli/src/rc-file.ts

```diff
@@ -25,27 +26,54 @@ type RawRCFile = {
     | null
 }
 
-export async function loadToolKitRC(logger: Logger, root: string, isAppRoot: boolean): Promise<RCFile> {
-  const result = await explorer.search(root)
+// yaml will automatically stringify any symbols in keys so just use strings
+// that won't be used normally
+export const toolKitOptionIdent = '__toolkit/option__'
+export const toolKitDefinedIdent = '__toolkit/defined__'
 
-  if (!result?.config) {
-    return emptyConfig
-  }
-  if (isAppRoot) {
-    rootConfig = result.filepath
-  } else if (result.filepath === rootConfig) {
-    // Make sure that custom plugins which don't have a config file won't cause
-    // the resolver to use the root config instead and start an infinite loop
-    // of config resolution.
-    logger.warn(
-      `plugin at ${s.filepath(path.dirname(root))} has no config file. please add an empty ${s.filepath(
-        '.toolkitrc'
-      )} file to avoid potential config resolution issues.`
-    )
-    return emptyConfig
+// minimally define the two custom tags so that yaml will parse them without
+// warning but will never be use them when stringifying
+const toolKitOption = {
+  identify: () => false,
+  tag: '!toolkit/option',
+  resolve: (value) => value
+} satisfies YAML.ScalarTag
+const toolKitDefined = {
+  identify: () => false,
+  tag: '!toolkit/defined',
```

Comment:
**suggestion(bikeshedding)**: i think `!toolkit/if-defined` (or maybe `if-option-defined`? bit verbose tho) would be clearer in intent and more readable